### PR TITLE
JSON schema file for YAML configuration schema validation

### PR DESCRIPTION
**Pull request by buddhikac96**
Fixes https://github.com/winsw/winsw/issues/537

### CODE REVIEW

File: doc/yamlConfigFile.md [hunk 1]
```diff
@@ -28,6 +28,13 @@ log:
     mode: roll
 ```
 
+## YAML configuration schema validation
+
+Users can validate YAML configurations file against JSON schema.
+First you should convert YAML file into JSON and then you can
```
Comment (oleg-nenashev) [inline doc/yamlConfigFile.md:34]: It is not needed if you use https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml . I would suggest documenting such use-case which is more friendly to users

File: doc/yamlConfigurationSchema.json [hunk 2]
```diff
@@ -0,0 +1,365 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "type": "object",
+  "title": "Windows Service Wrapper. YAML configuration schema",
+  "description": "This JSON schema validate the YAML configuaration file.",
```
Comment (oleg-nenashev) [inline doc/yamlConfigurationSchema.json:5]: Link the YAML config documentation page for more details?

Comment (oleg-nenashev) [inline doc/yamlConfigurationSchema.json:5]:
```suggestion
  "description": "This JSON schema validate the YAML configuration file.",
```

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
I have tested it with https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml , and it works well. There is an issue with boolean properties which are registered as String with boolPattern in the schema. Would be great to actually make them boolean if possible, but it is a subject for a separate discussion with @buddhikac96 and @NextTurn . Might be fine as well

Requesting changes because of the typo

**Review by oleg-nenashev — APPROVED** [review]
Fine with me, thanks for addressing the comments.
We can definitely start from that and improve in follow-ups

File: doc/yamlConfigurationSchema.json [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+          "$ref": "#/definitions/boolPattern"
+        },
+        "settings": {
+          "$id": "#/definitions/extensions/settings",
+          "type": "object"
+        }
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (oleg-nenashev) [reply doc/yamlConfigurationSchema.json:366]:
```suggestion
}

```

**Comment by buddhikac96** [comment]
> Fine with me, thanks for addressing the comments.
> We can definitely start from that and improve in follow-ups

Thanks. Are we willing to integrate YAML schema validation someday or we will keep it to use from VSCode?

File: doc/yamlConfigFile.md [hunk 4]
```diff
@@ -28,6 +28,13 @@ log:
...
+validate the JSON file against JSON schema.
+[Find the JSON schema in this link.](yamlConfigurationSchema.json)
```
[the remark this answers is not in the document]
Comment (NextTurn) [reply doc/yamlConfigFile.md]: After this link was removed the JSON file is not referenced anywhere.

[the remark this answers is not in the document]
Comment (buddhikac96) [reply doc/yamlConfigFile.md]: So how should I reference the file? Just use the file name without a link?

**Pull request merged**